src/charm.py

```diff
@@ -421,138 +371,57 @@ def get_all_hostnames(self) -> List[str]:
         relation = self._get_nginx_relation()  # type: ignore[arg-type]
         if not relation:
             return []
-        definition = self._get_definition_from_relation(relation)  # type: ignore[arg-type]
+        try:
+            definition = self._get_definition_from_relation(relation)  # type: ignore[arg-type]
+        except InvalidIngressError:
+            LOGGER.warning("Invalid ingress definition")
```

Comment:
I'm wondering if this should be an error (since for example in terraform we expected something with a relation but it did not get assigned whatever is expected)